### PR TITLE
HLRC: Split ILM request converters

### PR DESCRIPTION
**Pull request by AthenaEryma**
In an effort to encapsulate different clients, the request converters
are being shuffled around. This splits the Index Lifecycle Management
request converters.

This has already been done for pretty much all the other clients by
@hub-cap, ILM hasn't yet.

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleRequestConverters.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import org.elasticsearch.client.indexlifecycle.RemoveIndexLifecyclePolicyRequest;
+import org.elasticsearch.client.indexlifecycle.RetryLifecyclePolicyRequest;
+import org.elasticsearch.client.indexlifecycle.SetIndexLifecyclePolicyRequest;
+import org.elasticsearch.client.indexlifecycle.StartILMRequest;
+import org.elasticsearch.client.indexlifecycle.StopILMRequest;
+import org.elasticsearch.common.Strings;
+
+import java.io.IOException;
+
+public class IndexLifecycleRequestConverters {
```
Comment (talevy) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleRequestConverters.java]: Do we want to make this class final and make the empty constructor private so an instance of this class is never made?

**Review by talevy — CHANGES_REQUESTED** [review]
just left a few nits

File: client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleRequestConverters.java (hunk 1 above)
Comment (gwbrown) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleRequestConverters.java]: I think this is a good idea and we should probably do it to the other `*RequestConverters` classes - what do you think @hub-cap?

Comment (talevy) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleRequestConverters.java]: from sampling a few of the others, looks like a few of them do already

Comment (gwbrown) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleRequestConverters.java]: You're right, some do! I went ahead and made the changes here, I'll probably open another PR to do this consistently across the `*RequestConverters` classes as we should be consistent about this.

**Review by talevy — APPROVED** [review]
LGTM

**Comment by hub-cap** [comment]
yay!

File: client/rest-high-level/src/test/java/org/elasticsearch/client/IndexLifecycleRequestConvertersTests.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+import static org.elasticsearch.client.RequestConvertersTests.randomIndicesNames;
+import static org.elasticsearch.client.RequestConvertersTests.setRandomIndicesOptions;
+import static org.elasticsearch.client.RequestConvertersTests.setRandomMasterTimeout;
+import static org.elasticsearch.client.RequestConvertersTests.setRandomTimeoutTimeValue;
+import static org.elasticsearch.client.indexlifecycle.LifecyclePolicyTests.createRandomPolicy;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class IndexLifecycleRequestConvertersTests extends ESTestCase {
+
+
```
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/IndexLifecycleRequestConvertersTests.java]: so many spaces :)

File: client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java [hunk 3]
```diff
@@ -1469,143 +1457,6 @@ public void testEnforceSameContentType() {
...
-        RetryLifecyclePolicyRequest req = new RetryLifecyclePolicyRequest(indices);
-        Map<String, String> expectedParams = new HashMap<>();
-        setRandomMasterTimeout(req::setMasterTimeout, TimedRequest.DEFAULT_MASTER_NODE_TIMEOUT, expectedParams);
-        setRandomTimeoutTimeValue(req::setTimeout, TimedRequest.DEFAULT_ACK_TIMEOUT, expectedParams);
-        Request request = RequestConverters.retryLifecycle(req);
-        assertThat(request.getMethod(), equalTo(HttpPost.METHOD_NAME));
-        String idxString = Strings.arrayToCommaDelimitedString(indices);
-        assertThat(request.getEndpoint(), equalTo("/" + (idxString.isEmpty() ? "" : (idxString + "/")) + "_ilm/retry"));
-        assertThat(request.getParameters(), equalTo(expectedParams));
-    }
```
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java]: too many spaces

**Review by hub-cap — APPROVED** [review]
so awesome, ty @gwbrown!! Just 2 double space issues

**Pull request merged**